### DISCUSSION TITLE
IRSA support in PrivateBin Application

### DISCUSSION THREAD

**Discussion by rizwannadeemupwork** [PrivateBin/PrivateBin#1133] [Q&A]
Hello Team, 

we are looking for integration privatebin with S3 integration with IRSA method, at the moment we did the integration but using IAM user access.key and secret.key but we would like to move it to IAM roles and that requires IRSA for integration application support. 

So just wanted to know, do we have this option available in PrivateBin code? or by any changes this will be in your future release? 

Thanks,
Rizwan Nadeem

**Comment by rugk** [comment]
Hi, PrivateBin does _not_ have a concept of an authenticated user (or any user identity to start of), to start of and we have repeatedly made clear that this is not something we will add. See https://github.com/PrivateBin/PrivateBin/issues/657 for example.

Instead depending on what exactly you want to do (which is not clear from what you said), you may implement any identity management or something like that in your webserver. There are examples in the wiki and this FAQ may get you started (with easier authentication methods aka BasicAuth etc.):
https://github.com/PrivateBin/PrivateBin/wiki/FAQ#how-can-i-link-to-a-read-only-mode-where-users-cant-submit-pastes

**Comment by elrido** [comment]
Should this be a query on the authentication of the S3 storage model against your S3 bucket, we are mainly a consumer of the AWS SDK for PHP and may or may not support this if that library handles that for us (details in the [PHP SDK docs](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials.html)). The sample configurations outline two AWS scenarios that I've heard of:
https://github.com/PrivateBin/PrivateBin/blob/48079483be9e7f7b5f623a07f2fcb7f157928d51/cfg/conf.sample.php#L223-L244

I'm not an AWS user myself (or rather not anymore) and we have no project related resources in that environment, so can't really maintain that integration beyond frequently updating the library. We have to rely on the community to provide any fixes or extensions should they become necessary.
